File: school-finder-scraper/utils/logger.py
import logging
import os
import sys
from datetime import datetime

def setup_logger(log_level='INFO'):
    """
    Konfiguracja loggera dla scrapera
    
    Args:
        log_level (str): Poziom logowania (DEBUG, INFO, WARNING, ERROR, CRITICAL)
        
    Returns:
        logging.Logger: Skonfigurowany logger
    """
    # Mapowanie poziomów logowania
    log_levels = {
        'DEBUG': logging.DEBUG,
        'INFO': logging.INFO,
        'WARNING': logging.WARNING,
        'ERROR': logging.ERROR,
        'CRITICAL': logging.CRITICAL
    }
    
    # Ustawienie poziomu logowania
    level = log_levels.get(log_level.upper(), logging.INFO)
    
    # Utworzenie katalogu logs, jeśli nie istnieje
    logs_dir = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'logs')
    os.makedirs(logs_dir, exist_ok=True)
    
    # Nazwa pliku logu z datą
    log_filename = os.path.join(logs_dir, f"scraper_{datetime.now().strftime('%Y%m%d_%H%M%S')}.log")
    
    # Konfiguracja loggera
    logger = logging.getLogger('school_finder_scraper')
    logger.setLevel(level)
    
    # Handler dla konsoli
    console_handler = logging.StreamHandler(sys.stdout)
    console_handler.setLevel(level)
    console_format = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
    console_handler.setFormatter(console_format)
    
    # Handler dla pliku
    file_handler = logging.FileHandler(log_filename)
    file_handler.setLevel(level)
    file_format = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
    file_handler.setFormatter(file_format)
    
    # Dodanie handlerów do loggera
    logger.addHandler(console_handler)
    logger.addHandler(file_handler)
    
    return logger

File: src/lib/scraper/utils/logger.ts
/**
 * Logger dla scrapera
 */

export enum LogLevel {
  DEBUG = 'DEBUG',
  INFO = 'INFO',
  WARNING = 'WARNING',
  ERROR = 'ERROR'
}

export class Logger {
  private level: LogLevel;
  private prefix: string;
  
  constructor(level: LogLevel = LogLevel.INFO, prefix: string = 'Scraper') {
    this.level = level;
    this.prefix = prefix;
  }
  
  /**
   * Ustawienie poziomu logowania
   */
  setLevel(level: LogLevel): void {
    this.level = level;
  }
  
  /**
   * Logowanie na poziomie DEBUG
   */
  debug(message: string, ...args: any[]): void {
    if (this.shouldLog(LogLevel.DEBUG)) {
      console.debug(`[${this.prefix}] [DEBUG] ${message}`, ...args);
    }
  }
  
  /**
   * Logowanie na poziomie INFO
   */
  info(message: string, ...args: any[]): void {
    if (this.shouldLog(LogLevel.INFO)) {
      console.info(`[${this.prefix}] [INFO] ${message}`, ...args);
    }
  }
  
  /**
   * Logowanie na poziomie WARNING
   */
  warn(message: string, ...args: any[]): void {
    if (this.shouldLog(LogLevel.WARNING)) {
      console.warn(`[${this.prefix}] [WARNING] ${message}`, ...args);
    }
  }
  
  /**
   * Logowanie na poziomie ERROR
   */
  error(message: string, ...args: any[]): void {
    if (this.shouldLog(LogLevel.ERROR)) {
      console.error(`[${this.prefix}] [ERROR] ${message}`, ...args);
    }
  }
  
  /**
   * Sprawdzenie, czy dany poziom logowania powinien być wyświetlany
   */
  private shouldLog(level: LogLevel): boolean {
    const levels = {
      [LogLevel.DEBUG]: 0,
      [LogLevel.INFO]: 1,
      [LogLevel.WARNING]: 2,
      [LogLevel.ERROR]: 3
    };
    
    return levels[level] >= levels[this.level];
  }
}

/**
 * Utworzenie loggera z odpowiednim poziomem logowania
 */
export function setupLogger(level: string = 'INFO'): Logger {
  const logLevel = (level in LogLevel) ? (LogLevel as any)[level] : LogLevel.INFO;
  return new Logger(logLevel);
}